pkg/gofr/container/datasources.go

```diff
@@ -148,8 +150,39 @@ type CassandraBatch interface {
 	ExecuteBatchCAS(name string, dest ...any) (bool, error)
 }
 
-type CassandraProvider interface {
+type CassandraWithContext interface {
+	// QueryWithCtx executes the query with a context and binds the result into dest parameter.
+	// Accepts pointer to struct or slice as dest parameter for single and multiple rows retrieval respectively.
+	QueryWithCtx(ctx context.Context, dest any, stmt string, values ...any) error
+
+	// ExecWitCtx executes the query with a context, without returning any rows.
+	ExecWitCtx(ctx context.Context, stmt string, values ...any) error
+
+	// ExecCASWithCtx executes a lightweight transaction with a context.
+	ExecCASWithCtx(ctx context.Context, dest any, stmt string, values ...any) (bool, error)
+
+	// NewBatchWithCtx creates a new Cassandra batch with context.
+	NewBatchWithCtx(ctx context.Context, name string, batchType int) error
+
 	Cassandra
+	CassandraBatchWithContext
+}
+
+type CassandraBatchWithContext interface {
+	// BatchQueryWithCtx adds the query to the batch operation with a context.
+	BatchQueryWithCtx(ctx context.Context, name, stmt string, values ...any) error
+
+	// ExecuteBatchWithCtx executes a batch operation with a context.
+	ExecuteBatchWithCtx(ctx context.Context, name string) error
+
+	// ExecuteBatchCASWithCtx executes a batch operation with context and returns the result.
+	ExecuteBatchCASWithCtx(ctx context.Context, name string, dest ...any) (bool, error)
+
+	CassandraBatch
```

Comment:
This is not required as `Cassandra` interface already embeds the CassandraBatch

Reply:
Sure removed